builtin/providers/aws/resource_aws_spot_datafeed_subscription.go

```diff
@@ -0,0 +1,93 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsSpotDataFeedSubscription() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSpotDataFeedSubscriptionCreate,
+		Read:   resourceAwsSpotDataFeedSubscriptionRead,
+		Delete: resourceAwsSpotDataFeedSubscriptionDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"bucket": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"prefix": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsSpotDataFeedSubscriptionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	params := &ec2.CreateSpotDatafeedSubscriptionInput{
+		Bucket: aws.String(d.Get("bucket").(string)),
+	}
+
+	if v, ok := d.GetOk("prefix"); ok {
+		params.Prefix = aws.String(v.(string))
+	}
+
+	log.Printf("[INFO] Creating Spot Datafeed Subscription")
+	_, err := conn.CreateSpotDatafeedSubscription(params)
+	if err != nil {
+		return errwrap.Wrapf("Error Creating Spot Datafeed Subscription: {{err}}", err)
+	}
+
+	d.SetId("spot-datafeed-subscription")
```

Comment:
Actually my comment here is in correct, I forgot they were keyed in state by the user-supplied name rather than the ID.